xarray/core/dataset.py

```diff
@@ -2884,16 +2884,24 @@ def _validate_interp_indexer(x, new_x):
                 )
             return x, new_x
 
+        validated_indexers = {
+            k: _validate_interp_indexer(maybe_variable(obj, k), v)
+            for k, v in indexers.items()
+        }
+
+        if method in ["linear", "nearest"]:
+            for k, v in validated_indexers.items():
```

Comment:
not sure if we need `decompose_interp` here.

@pums974 could you please take a look if you have time?

Reply:
Hopefully this is not just me having broken my xarray version but the `decompose_interp` change seems to have changed the `fill_value` behavior as well, compare the examples and results from https://github.com/pydata/xarray/issues/3962. Now we only do 1D interpolation so `scipy.interpolate.interp1d` is always in use. At least the api is consistent now! And we support far more interpolation methods this way, whether it works correctly though I don't know.

But I can't figure out how I can get the nearest interpolated value when extrapolting now.

It is also interesting that `NumpyInterpolator` (which has my favorite default settings; linear interpolation and nearest interpolated extrapolation) rewrites the numpy API to follow `scipy.interpolate.interp1d` but we don't bother doing that with with any other interpolation class even though the scipy API is notoriously inconsistent.

Reply:
This change is the only one I'm a little unsure about. It is also mutating the dict while looping over the same dict, which is not nice.

@Illviljan can you take a deeper look at it and see whether we need `decompose_interp` here?

Reply:
@dcherian, could you remind me how `decompose_interp` relates to using `missing._localize` in these lines? 
If I understand this part correctly we're just removing values outside of the interpolation range, right? And `decompose_interp` seems to only trigger later when we call `missing.interp` on each variable.

If so, isn't the `decompose_interp` discussion unrelated to these performance enhancements?

Reply:
Thanks @Illviljan that makes sense.